OpenSim/Common/Component.cpp

```diff
@@ -319,6 +331,11 @@ void Component::disconnect()
 
 void Component::addToSystem(SimTK::MultibodySystem& system) const
 {
+    // If being asked to be added to the same System that it is already
+    // a part, there is nothing to be done.
+    if (hasSystem() && (&getSystem() == &system)) {
+        return;
+    }
```

Comment:
Thanks for the explanation. I need to read more of the PR and keep thinking before giving an opinion.